test/integration_test.go

```diff
@@ -404,6 +409,35 @@ func (ts *IntegrationTestSuite) TestActivityCancelRepro() {
 	ts.EqualValues(expected, ts.activities.invoked())
 }
 
+func (ts *IntegrationTestSuite) TestCancelActivity() {
+	var expected []string
+	err := ts.executeWorkflow("test-cancel-activity", ts.workflows.CancelActivity, &expected)
+	ts.NoError(err)
+	ts.EqualValues(expected, ts.activities.invoked())
+}
+
+func (ts *IntegrationTestSuite) TestCancelTimer() {
+	var expected []string
+	err := ts.executeWorkflow("test-cancel-timer", ts.workflows.CancelTimer, &expected)
+	ts.NoError(err)
+	ts.EqualValues(expected, ts.activities.invoked())
+}
+
+func (ts *IntegrationTestSuite) TestCancelChildWorkflow() {
+	var expected []string
+	err := ts.executeWorkflow("test-cancel-child-workflow", ts.workflows.CancelChildWorkflow, &expected)
+	ts.NoError(err)
+	ts.EqualValues(expected, ts.activities.invoked())
+}
+
+func (ts *IntegrationTestSuite) TestCancelActivityImmediately() {
+	ts.T().Skip(`Currently fails with "PanicError": "unknown decision internal.decisionID{decisionType:0, id:"5"}, possible causes are nondeterministic workflow definition code or incompatible change in the workflow definition`)
```

Comment:
This test doesn't pass now. @mfateev, you can use it to test immediate cancelation.